crates/rattler_conda_types/src/version/bump.rs

```diff
@@ -0,0 +1,31 @@
+use thiserror::Error;
+
+/// VersionBumpType is used to specify the type of bump to perform on a version.
+#[derive(Clone)]
+pub enum VersionBumpType {
+    /// Bump the major version number.
+    Major,
+    /// Bump the minor version number.
+    Minor,
+    /// Bump the patch version number.
+    Patch,
+    /// Bump the last  version number.
+    Last,
```

Comment:
Can you add a `Segment(uint)` type? So that we can explicitly select the segment to bump?

We might also consider that bumping non-existent segments should be possible. Usually, all non-existent segments are implicit `.0`, so bumping the `patch` of a `1.0` could also result in `1.0.1`.

Reply:
> Can you add a Segment(uint) type? So that we can explicitly select the segment to bump?

Good idea. Will do.

Reply:
> We might also consider that bumping non-existent segments should be possible. Usually, all non-existent segments are implicit .0, so bumping the patch of a 1.0 could also result in 1.0.1.

Don't you prefer to instead fail, so it also builds incentive for the user to explicitly use `x.x.x` instead of `x` or `x.x`?

Unless you feel strongly here, I would prefer to let it for another PR (I can open a ticket if you want).

Reply:
Added support for `VersionBumpType::Segment(i32)` in 9283770893b34b895694b1b28652ffbb618db55f (rust side only).